drivers/adc/adc_context.h

```diff
@@ -181,13 +183,16 @@ static inline void adc_context_complete(struct adc_context *ctx, int status)
 static inline void adc_context_start_read(struct adc_context *ctx,
 					  const struct adc_sequence *sequence)
 {
-	ctx->sequence = sequence;
+	(void)memcpy(&ctx->sequence, sequence, sizeof(struct adc_sequence));
 	ctx->status = 0;
 
-	if (ctx->sequence->options) {
+	if (sequence->options) {
+		(void)memcpy(&ctx->options, sequence->options,
+			     sizeof(struct adc_sequence_options));
```

Comment:
Why not just use `ctx->options = sequence->options;` and let the compiler decide how to handle the copying? Especially when the structure is not big, it is likely that this could be optimized.
Same story above, with `ctx->sequence`.